src/Surfnet/StepupMiddleware/ApiBundle/Identity/Repository/AuthorizationRepository.php

```diff
@@ -73,12 +84,38 @@ public function getInstitutionsForRole(InstitutionRole $role, IdentityId $actorI
         );
 
         $institutions = $qb->getQuery()->getArrayResult();
-
-        $result = new InstitutionCollection();
         foreach ($institutions as $institution) {
+            $this->logger->notice(
+                sprintf('Adding %s to authorized institutions', $institution['institution'])
+            );
             $result->add(new Institution((string)$institution['institution']));
         }
 
+        // Also get the institutions that are linked to the user via the 'institution_relation' field.
+        // Effectively getting the use_raa relation.
+        // See https://www.pivotaltracker.com/story/show/181537313
+        $qb = $this->_em->createQueryBuilder()
```

Comment:
Would it be sufficient to fix the where clause in the query above?

Reply:
I have tried this after your suggestion. And added an additional join condition to the existing query. That did not result in the correct outcome.

For some extra context, query 1:
Selects the authorized institutions that apply for the home institution of the RA that is interacting with Middleware (the ra_institution)

And query 2:
Checks the Institution the RA has been selected for (using the SELECT_RA feature). And selects in which institutions the RA has RA(A) rights.

Reply:
:+1: thanks for validating!